### DISCUSSION TITLE
How can I impersonate a User in PNP Powershell on a Azure Function called by an SPFX WebPart

### DISCUSSION THREAD

**Discussion by HyroHyKen** [pnp/powershell#1765] [Q&A]
Hello!
I have made a WebPart in SPFX in order to call an Azure Function from Sharepoint

This Azure function is in PowerShell and I need to call the PnP Powershell function from there 

My main goal is to impersonate the user that is calling the function from the web part.

Previously, I could connect properly to PNP services with hardcoded user credentials. 
I tried to connect with "Connect-PnpOnline" with the Token generated from the AadTokenProviderFactory by my WebPart every Pnp's function that I called sends me back a "401 Unauthorized"

I've succeeded to do so with Microsoft Graph's Modules but I can't figure out how to make Pnp's access token work

Can you explain to me how to use Connect-PnPOnline and AccessToken to make impersonated PnP Powershell calls in my azure function?
Is there a different way to retrieve the expected token?

Thanks for your help.

**Comment by JMTeamway** [comment]
I'm very interested too in that scenario !

**Comment by JMTeamway** [comment]
It seems every attempt I did to use Connect-PnpOnline with the token parameters fails.

Even if I successfully connect using credentials, and retrieve token with Get-PnPAccessToken or GetPnPAppAuthAccessToken, those tokens do not allow me to perform a Connect-PnPOnline.

I wonder if his feature is really implemented and functionnal ...

**Comment by JMTeamway** [comment]
Hi @erwinvanhunen , 
I noticed some interesting post of you ( https://github.com/pnp/powershell/issues/305 ) related to this command.

Maybe you could help us to understand wether what we want is achievable, and how we should do things ?

**Reply by rvdwegen** [reply]
@JMTeamway Hi, did you ever get AccessToken working?

**Reply by JMTeamway** [reply]
Well, depends on working with what !
I correctly uses AccessToken with Microsoft Graph Module.

The issue is to use Pnp Modules, and specially its Connect-PnpOnline command with Token parameter.

**Reply by rvdwegen** [reply]
Yea same haha. I've now got everything we need working (AzureAD, MgGraph, ExchangeOnlineManagement, Az) except PnP
